### PR TITLE
solve submodules issues

### PR DESCRIPTION
**Pull request by jcolomb**
Please use the review process to complete the debugging:
- [x] submodules are added to team
- [ ] submodule templates that is cloned are not pushed to newly created repos (l.273 probably)
- [x] gitmodules to point to the newly created repos (and not the template submodule)

Issues with committing changes in the submodule or initiating gin. Currently using a workaround using the initiation scripts to add and push content to the created repositories (need ssh). 

submodules cannot be pushed because they are no gin repo. We would need a way to push submodule content, something equivalent to `git submodule for each gin init` after creating them.

NB: since on gindata the link with ssh submodule do not work, we use relative links.

### CODE REVIEW

File: utonics/labproject/main.go [hunk 1]
```diff
@@ -266,16 +266,20 @@ func newProject(values map[string][]string, botClient, userClient *worker.Client
 		return msgs, err
 	}
 
-	for _, submodule := range submodules {
+	for smName, submodule := range submodules {
 		os.Chdir(submodule.path)
+		smName = strings.ReplaceAll(smName, "/", "_")
 		msgs = append(msgs, "Uploading submodule to new project repository")
-		if err := uploadProjectRepository(botClient, remoteName); err != nil {
+		if err := uploadProjectRepository(botClient, project + "." + smName); err != nil {
```
Comment (achilleas-k) [inline utonics/labproject/main.go:273]: This argument is meant to be the name of the remote for upload.  the argument is used to run (for the default remote name `origin` and branch `master`):
```
git push origin master
git annex copy --to=origin
```

Changing it to the repository name will fail, unless we're also changing remote names, but that doesn't seem to be part of this patch.

File: utonics/labproject/main.go [hunk 2]
```diff
@@ -325,6 +325,7 @@ func newProject(values map[string][]string, botClient, userClient *worker.Client
 		return msgs, err
 	}
 	for smName := range submodules {
+	  smName = strings.ReplaceAll(smName, "/", "_")
```
Comment (achilleas-k) [inline utonics/labproject/main.go:332]: Something's wrong with the indentation here.  Is this indented with spaces?
You can use `gofmt` to automatically format code.

**Comment by jcolomb** [comment]
doing manual editing:

`git submodule set-url 06_disseminations/02_manuscript "ssh://git@gindata.biologie.hu-berlin.de:10022/testtonic/test003_newcode.06_disseminations_02_manuscript.git"`
this works, but the link on gindata does not work because of the 10022 port issue

`git submodule set-url 06_disseminations/02_manuscript "https://gindata.biologie.hu-berlin.de/testtonic/test003_newcode.06_disseminations_02_manuscript"`
This solves the link, but the repo is still empty and the sync scripts does not do anything.

File: utonics/labproject/main.go (hunk 2 above)
Comment (jcolomb) [reply utonics/labproject/main.go]: I hope gofmt worked.

**Comment by jcolomb** [comment]
solved an issue with os-chdir not getting to the right folder, but it is still not working.
Adding one submodule  (labcommons) always worked, I forgot to commit the change before the push.
 
 git submodule set-url:
 - was not done in parent repo earlier, it drived no error
 - now it brings an error (was not there before)
 `Uploading submodule to new project repository
Upload failed: git-annex: First run: git-annex init `
- the .gitmodule file is still showing the previous url, it may need to be added directly (running `git add .gitmodules`)
